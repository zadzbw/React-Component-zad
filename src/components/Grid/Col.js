/**
 * Created by zad on 16/12/8.
 */
import React, {Component, PropTypes} from 'react';
import classNames from 'classnames';
import _forEach from 'lodash/forEach';

import './Col.less';

const colPrefix = 'zad-col';

export default class Col extends Component {
  static propTypes = {
    span: PropTypes.number,
    offset: PropTypes.number,
    xs: PropTypes.number,
    sm: PropTypes.number,
    md: PropTypes.number,
    lg: PropTypes.number
  };

  static defaultProps = {
    span: 0,
    offset: 0
  };

  render() {
    const {span, offset, xs, sm, md, lg, className, children, ...props} = this.props;
    const sizeMap = {
      xs: xs,
      sm: sm,
      md: md,
      lg: lg
    };

    let sizeClassObj = {};
    _forEach(sizeMap, (value, key) => {
      sizeClassObj = Object.assign({}, sizeClassObj, {
        [`${colPrefix}-${key}-${value}`]: !!value
      });
    });

    const colClass = classNames(className, {
      [`${colPrefix}-${span}`]: !!span,
      [`${colPrefix}-offset-${offset}`]: !!offset
    }, sizeClassObj);

    return (
      <div {...props} className={colClass}>
        {children}
      </div>
    );
  }
}
