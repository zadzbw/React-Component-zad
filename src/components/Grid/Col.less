@import "../../style/variables";
@import "./GridMixin";

.init-grid-columns();
.make-grid();
.make-grid(-xs);

@media (min-width: @screen-sm-min) {
  .make-grid(-sm);
}

@media (min-width: @screen-md-min) {
  .make-grid(-md);
}

@media (min-width: @screen-lg-min) {
  .make-grid(-lg);
}
