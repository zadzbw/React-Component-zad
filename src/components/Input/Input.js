/**
 * Created by zad on 17/3/23.
 */
import React, {Component, PropTypes} from 'react';

export default class Input extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <input type="text"/>
    );
  }
}
