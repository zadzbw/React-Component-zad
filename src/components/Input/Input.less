@import "../../style/variables";
@import "../../style/mixins/index";

.zad-input {
  position: relative;
  display: inline-block;
  width: 100%;
  height: @input-height-base;
  font-size: @font-size-sm;
  line-height: @line-height-base;
  padding: @input-padding-vertical-base @input-padding-horizontal;
  color: @input-color;
  background: @input-bg none;
  border: 1px @input-border-color solid;
  border-radius: @border-radius-base;
  cursor: text;
  transition: all .3s;
  .placeholder(); // Reset placeholder

  &:hover {
    border-color: tint(@primary-color, 20%);
  }

  &:focus {
    border-color: tint(@primary-color, 20%);
    box-shadow: 0 0 0 2px fade(@primary-color, 20%);
    outline: none;
  }

  &[disabled] {
    background-color: @input-disabled-bg;
    opacity: 1;
    cursor: not-allowed;
    color: @disabled-color;
    &:hover {
      border-color: @input-border-color
    }
  }

  &-sm {
    padding: @input-padding-vertical-sm @input-padding-horizontal;
    height: @input-height-sm;
  }

  &-lg {
    padding: @input-padding-vertical-lg @input-padding-horizontal;
    height: @input-height-lg;
  }
}
