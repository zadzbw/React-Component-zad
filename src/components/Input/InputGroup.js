/**
 * Created by zad on 17/3/23.
 */
import React, {Component, PropTypes} from 'react';
import classNames from 'classnames';
import isOneOf from '../../utils/isOneOf';

const _type = {
  size: ['small', 'large']
};
const inputGroupPrefix = 'zad-input-group';

export default class InputGroup extends Component {
  constructor(props) {
    super(props);
  }

  static propTypes = {
    size: PropTypes.string,
    compact: PropTypes.bool,
  };

  static defaultProps = {
    size: 'default',
    compact: true,
  };

  render() {
    const {size, compact, className, children, ...props} = this.props;
    const sizeSuffix = ({
      large: 'lg',
      small: 'sm',
    })[size];

    const inputGroupClass = classNames(inputGroupPrefix, className, {
      [`${inputGroupPrefix}-${sizeSuffix}`]: size && isOneOf(_type.size, size),
      [`${inputGroupPrefix}-compact`]: compact,
    });

    return (
      <span {...props} className={inputGroupClass}>
        {children}
      </span>
    );
  }
}
