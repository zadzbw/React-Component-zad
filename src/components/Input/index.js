/**
 * Created by zad on 17/3/23.
 */
import './Input.less';

import Input from './Input';
import InputGroup from './InputGroup';
Input.Group = InputGroup;

export default Input;
