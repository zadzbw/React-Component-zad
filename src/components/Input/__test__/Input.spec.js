/**
 * Created by zad on 17/4/11.
 */
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Input from '../Input';
import InputGroup from '../InputGroup';
import Col from '../../../components/Grid/Col';

describe('Input Test', () => {
  it('normal', () => {
    class Test extends React.Component {
      constructor(props) {
        super(props);
        this.state = {input: 'test1'};
      }

      testInput = (e) => {
        this.setState({
          input: e.target.value
        })
      };

      render() {
        return (
          <div>
            <Input disabled placeholder="disabled"/>
            <Input size={'large'} placeholder="large size"/>
            <Input placeholder="default size" onChange={this.testInput} value={this.state.input}/>
            <Input size={'small'} placeholder="small size"/>
          </div>
        );
      }
    }

    const wrapper = shallow(<Test/>);

    // simulate change
    expect(wrapper.find(Input).at(2).props().value).toBe('test1');
    wrapper.find(Input).at(2).simulate('change', {keyCode: 13, target: {value: 'test2'}});
    expect(wrapper.find(Input).at(2).props().value).toBe('test2');

    expect(wrapper.find(Input).at(1).shallow().hasClass('zad-input-lg')).toBeTruthy();
    expect(wrapper.find(Input).at(3).shallow().hasClass('zad-input-sm')).toBeTruthy();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('InputGroup Test', () => {
  it('no compact', () => {
    const wrapper = shallow(
      <InputGroup size={'large'}>
        <Col span={6}>
          <Input defaultValue="0571"/>
        </Col>
        <Col span={8}>
          <Input defaultValue="26888888"/>
        </Col>
      </InputGroup>
    );

    expect(wrapper.hasClass('zad-input-group-lg')).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('no compact', () => {
    const wrapper = shallow(
      <InputGroup size={'small'} compact={true}>
        <Input style={{width: '15%'}} defaultValue="+86"/>
        <Input style={{width: '20%'}} defaultValue="0571" size={'small'}/>
        <Input style={{width: '30%'}} defaultValue="26888888"/>
      </InputGroup>
    );

    expect(wrapper.hasClass('zad-input-group-sm')).toBeTruthy();
    expect(wrapper.hasClass('zad-input-group-compact')).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
