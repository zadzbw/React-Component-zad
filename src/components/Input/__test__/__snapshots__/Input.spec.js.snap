exports[`Input Test normal 1`] = `
<div>
  <Input
    disabled={true}
    onPressEnter={[Function]}
    placeholder="disabled"
    size="default"
    type="text" />
  <Input
    onPressEnter={[Function]}
    placeholder="large size"
    size="large"
    type="text" />
  <Input
    onChange={[Function]}
    onPressEnter={[Function]}
    placeholder="default size"
    size="default"
    type="text"
    value="test2" />
  <Input
    onPressEnter={[Function]}
    placeholder="small size"
    size="small"
    type="text" />
</div>
`;

exports[`InputGroup Test no compact 1`] = `
<span
  className="zad-input-group zad-input-group-lg">
  <Col
    offset={0}
    span={6}>
    <Input
      defaultValue="0571"
      onPressEnter={[Function]}
      size="default"
      type="text" />
  </Col>
  <Col
    offset={0}
    span={8}>
    <Input
      defaultValue="26888888"
      onPressEnter={[Function]}
      size="default"
      type="text" />
  </Col>
</span>
`;

exports[`InputGroup Test no compact 2`] = `
<span
  className="zad-input-group zad-input-group-sm zad-input-group-compact">
  <Input
    defaultValue="+86"
    onPressEnter={[Function]}
    size="default"
    style={
      Object {
        "width": "15%",
      }
    }
    type="text" />
  <Input
    defaultValue="0571"
    onPressEnter={[Function]}
    size="small"
    style={
      Object {
        "width": "20%",
      }
    }
    type="text" />
  <Input
    defaultValue="26888888"
    onPressEnter={[Function]}
    size="default"
    style={
      Object {
        "width": "30%",
      }
    }
    type="text" />
</span>
`;
